### PR TITLE
Update CdoFieldDropdown validation of initial values

### PR DESCRIPTION
**Pull request by fisher-alice**
This PR updates the overridden`doClassValidation_` to handle valid initial values from xml, whether or not the initial values have surrounding quotes.

Currently, if a block field has an initial value set without quotes, for example, 'spinning right' below:

```
  <block type="Poetry_startBehavior">
    <title name="ANIMATION_NAME">"twinkling_star"</title>
    <title name="BEHAVIOR">spinning_right</title>
  </block>
```
then there is a warning in the dev console:

Screen Shot 2023-05-19 at 10 36 57 AM

Instead of 'spinning right', 'fluttering' is selected since it's the first option in the list of options:

Screen Shot 2023-05-19 at 10 31 33 AM

The reason is that in the block config, the initial value, in this case, 'spinning right', is compared with the list of options which contain a second pair of quotes so that the initial value must be the string '"spinning right"', not 'spinning right' in order to be read correctly:

Screen Shot 2023-05-19 at 10 39 17 AM

In [this commit](https://github.com/code-dot-org/code-dot-org/pull/51992/commits/f2e996a8baafab842d63f686b90718312125a404), if the initial value didn't have surrounding quotes, I added them, and then called on `super.doClassValidation_`. However, this caused a warning for the 'Poetry_whenLineShows' block for the 'N' field. 
```
  <block type="Poetry_whenLineShows">
    <title name="N">0</title>
  </block>
```
Since the block config lists options without an extra pair of quotes:

Screen Shot 2023-05-19 at 10 25 25 AM

Because the xml block configuration is not consistent for dropdown selections, I chose an implementation that first checks if the `newValue` is the same as an option and if not, check if adding quotes to it will get a match. If so, then reassign `newValue` with the additional quotes and mark it as a valid option.

## Testing story
I tested update locally on [this poetry levels](http://localhost-studio.code.org:3000/s/poetry-2021/lessons/1/levels/6).

Before:

Screen Shot 2023-05-19 at 10 51 20 AM

After: There were no warnings in dev console referring to the dropdown's value.

Screen Shot 2023-05-19 at 10 50 01 AM

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/blockly/addons/cdoFieldDropdown.js [hunk 1]
```diff
@@ -7,14 +7,42 @@ import {
...
+   * by quotes in xml.
    */
   doClassValidation_(newValue) {
     if (newValue === EMPTY_OPTION) {
       return newValue;
     } else {
-      return super.doClassValidation_(newValue);
+      const options = this.getOptions(true);
+      let isValueValid = false;
+      options.forEach(option => {
```
Comment (bencodeorg) [inline apps/src/blockly/addons/cdoFieldDropdown.js]: One cleanup idea: could we exit the loop early with something like:

```
for (const option of this.getOptions(true)) {
  if (option[1] === newValue) {
    return newValue;
  } else if (option[1] === `"${newValue}"`) {
    return `"${newValue}"`;
  }
}
```

Then you can drop the outside `if (!isValueValid)` check below and just return null if neither of these cases are hit?

**Review by bencodeorg — APPROVED** [review]
One idea to exit your loop early when possible but LGTM!

File: apps/src/blockly/addons/cdoFieldDropdown.js (hunk 1 above)
Comment (bencodeorg) [reply apps/src/blockly/addons/cdoFieldDropdown.js]: Also you may already know this, but it has tripped me up before -- you can't early return inside of a forEach in JS:

https://stackoverflow.com/questions/34653612/what-does-return-keyword-mean-inside-foreach-function

Comment (fisher-alice) [reply apps/src/blockly/addons/cdoFieldDropdown.js]: Thanks - definitely cleans things up!

**Review by mikeharv — APPROVED** [review]
Thanks for implementing this fix!

**Pull request merged**